compiler/noirc_frontend/src/node_interner.rs

```diff
@@ -228,7 +228,9 @@ pub struct NodeInterner {
 /// ```
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
 pub enum DependencyId {
+    Module(ModuleId),
```

Comment:
It seems `DependencyId` already existed in the compiler before introducing LSP, and for LSP it's used to track references... for example a `Variable` might reference a `Struct`, etc.

In this PR I added two new dependencies: `Module` and `Trait`. This is because in this path:

```rust
foo::bar::Baz::new
```

we can track that `foo` points to a `Module`, same for `Bar`, then `Baz` points to a struct (or a trait), etc. With this, clicking `foo` would take you to the `foo` module (this isn't implemented yet, though! But I needed this so that clicking `Baz` in the middle of the path works)

I'm wondering if a different type to track references might be better... something like `pub enum ReferenceId` that would be the same as `DependencyId` except that it also has `Module` and `Trait`... mainly to avoid ignoring `Module` and `Trait` in pattern matches around `dependency_graph`... what do you think?